### PR TITLE
Start-refactor-drops

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Refactoring
- [x] Part of https://github.com/kodadot/nft-gallery/issues/9575 

## Needs QA check
@prury 

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

### CODE REVIEW

**Comment by daiagi** [comment]
Hi @preschian @Jarsen136 

in this PR we mostly have
1. make composables be stand alone / no rely on arguments, so that they can be used more freely
2. components get what info they need straight from composables, instead of props
3. some shared state moved to store


overall effect is to reduce props drilling

after this comes refactor of paidGenerative, which is commented out for the moment

**Comment by prury** [comment]
can't access any of them, what am i missing? 

image

**Comment by prury** [comment]
Sailwings button disabled even if i have the amount and drop is open: 

image

**Comment by prury** [comment]
counter on refraction drop(30 already minted): 

image

**Comment by prury** [comment]
chevron on ended drops(chroma) specifying requisites for drop: 

image

canary: 

image

**Comment by prury** [comment]
holder of drop requirements(reflection of the sound) 

image

canary: 

image

**Comment by prury** [comment]
general drop status for all drops:
image

Allowlist, Public mint(open), Public mint(ended)

**Comment by daiagi** [comment]
@prury 
all issues you'vs found are fixed

File: stores/drop.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    isCaptutingImage: false,
+    selectedImage: '',
+    runtimeMintCount: 0,
+    drop: null,
+    mintedDropCount: null,
+  }),
+  getters: {
+    // loading: (state) => state.loading,
+    // walletConnecting: (state) => state.walletConnecting,
+    // isCaptutingImage: (state) => state.isCaptutingImage,
```
Comment (Jarsen136) [inline stores/drop.ts:27]:
```suggestion
```

**Review by Jarsen136 — APPROVED** [review]
code lgtm

**Comment by daiagi** [comment]
@prury 
there is a bug in reflection drop when minimum funds stuck on loading

image



unless you find more bugs, let's merge and solve this one in follow up

File: components/collection/drop/utils.ts [hunk 2]
```diff
@@ -3,3 +3,17 @@ export const getFakeEmail = () => {
...
+    email: getFakeEmail(),
+    hash,
+    address: accountId.value,
+    image: selectedImage.value,
+    metadata: metadata,
+  }
+
+  const response = await allocateCollection(body, drop.value.id)
+  return { raffleId: parseInt(response.result.id) }
+}
```
Comment (preschian) [inline components/collection/drop/utils.ts:19]: Unfinished? Can't find name `getImageInfo, selectedImage, accountId, selectedImage, allocateCollection, drop`. Could you revert it instead and add it to https://github.com/kodadot/nft-gallery/issues/9635?

File: components/collection/drop/Generative.vue [hunk 3]
```diff
@@ -256,7 +133,7 @@ const allocateRaffle = async (): Promise<{ raffleId: number }> => {
     metadata: metadata,
   }
 
-  const response = await allocateCollection(body, props.drop.id)
+  const response = await allocateCollection(body, drop.value.id)
```
Comment (preschian) [inline components/collection/drop/Generative.vue:136]: 'drop.value' is possibly 'null'

File: components/collection/unlockable/UnlockableCollectionBanner.vue [hunk 4]
```diff
@@ -27,38 +28,35 @@
...
 
 const description = computed(() => {
-  switch (props.drop?.type) {
+  switch (drop.value?.type) {
     case 'paid':
       return unlockableDesc(50)
-    case 'drop':
-      return unlockableDesc(40)
+
+    // case 'drop':
```
Comment (preschian) [inline components/collection/unlockable/UnlockableCollectionBanner.vue:58]: safe to remove

File: composables/party/useCursorDropEvents.ts [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
 
-export default (
-  drop: string,
-  mintingWatch?: Ref<boolean>[],
-  mintedNft?: Ref<DropMintedNft | undefined>,
-) => {
-  const { sendMessage } = useParty({ room: computed(() => drop) })
+export default (mintingWatch?: Ref<boolean>[]) => {
+  const drop = useDrop()
+  const { mintedNft } = useGenerativeDropMint()
```
Comment (preschian) [inline composables/party/useCursorDropEvents.ts:10]: should be like this?
```suggestion
  const { drop } = useDrop()
```

**Review by preschian — COMMENTED** [review]
please fix deepscan issue

**Comment by prury** [comment]
See listing disabled on old drops: 
/ahp/drops/alchemy
image

**Comment by Temidayo32** [comment]
👀

**Comment by yangwao** [comment]
@daiagi is it good to go?

**Comment by vikiival** [comment]
> @daiagi is it good to go?

Changes from @preschian are not implemented yet
Base stuff does not work as pointed out by @prury

File: components/collection/drop/utils.ts (hunk 2 above)
Comment (daiagi) [reply components/collection/drop/utils.ts:19]: hey

you maybe looked before i synced with main branch
this file is untouched by this PR

image

File: components/collection/drop/Generative.vue (hunk 3 above)
Comment (daiagi) [reply components/collection/drop/Generative.vue:139]: done

File: components/collection/unlockable/UnlockableCollectionBanner.vue (hunk 4 above)
Comment (daiagi) [reply components/collection/unlockable/UnlockableCollectionBanner.vue]: done

File: composables/party/useCursorDropEvents.ts (hunk 5 above)
Comment (daiagi) [reply composables/party/useCursorDropEvents.ts:10]: it should.
thank you 


done

**Comment by daiagi** [comment]
@prury 
check again plz

File: components/collection/drop/GenerativePreview.vue [hunk 6]
```diff
@@ -85,40 +76,44 @@
...
   holderOfCollection?: HolderOfCollectionProp
 }>()
 
+const { drop } = useDrop()
+const dropStore = useDropStore()
+
+const { maxCount, mintedCount, mintCountAvailable } = useGenerativeDropMint()
+const { mintedAmountForCurrentUser } = useCollectionEntity()
+
+watch(
```
Comment (preschian) [inline components/collection/drop/GenerativePreview.vue:98]: safe to remove? or intended for debugging purposes?
can use `watchEffect` here?

File: components/collection/drop/MintButton.vue [hunk 7]
```diff
@@ -6,91 +6,156 @@
...
-  withLabel?: boolean
-}>(() => {
-  if (isMintedOut.value) {
-    return {
-      label: $i18n.t('mint.unlockable.seeListings'),
-      disabled: false,
-    }
-  }
+  // Determine if there are available NFTs for minting
+  const hasNFTsAvailable = availableNfts.amount !== 0
```
Comment (preschian) [inline components/collection/drop/MintButton.vue:70]: `availableNfts` coming from?

**Review by preschian — APPROVED** [review]
otherwise code lgtm

**Comment by daiagi** [comment]
@preschian 
fixed

thank you for excellent code review

File: components/collection/drop/GenerativePreview.vue (hunk 6 above)
Comment (daiagi) [reply components/collection/drop/GenerativePreview.vue]: debug indeed
removed

File: components/collection/drop/MintButton.vue (hunk 7 above)
Comment (daiagi) [reply components/collection/drop/MintButton.vue]: bless you for spotting this

fixed

**Comment by yangwao** [comment]
Thanks @daiagi hopefully it on good path to improve flawless drops!
pay 100 usd

@prury can merge afterwards

**Comment by yangwao** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 10.35 USD/DOT ~ 9.662 $DOT
🧗 [1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8](https://polkadot.subscan.io/account/1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)
🔗 [0xef19d85059b260d19f33c4aa34459dc5efd366c1841606ec4de0664c3d1c97dd](https://polkadot.subscan.io/extrinsic/0xef19d85059b260d19f33c4aa34459dc5efd366c1841606ec4de0664c3d1c97dd)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by prury** [comment]
nice work! majority of things seems to be working fine
I'll keep communicating with @daiagi and testing it further

**Pull request merged**